src/wallet/rpcwallet.cpp

```diff
@@ -2768,10 +2769,37 @@ UniValue listwallets(const JSONRPCRequest& request)
 
         LOCK(pwallet->cs_wallet);
 
-        obj.push_back(pwallet->GetName());
+        loaded.push_back(pwallet->GetName());
+        loaded_set.insert(pwallet->GetName());
     }
 
-    return obj;
+    UniValue available(UniValue::VARR);
+    fs::path walletdir = GetWalletDir();
+    const unsigned char bdb_magic[4] = {0x62, 0x31, 0x05, 0x00}; // Berkeley DB Btree magic bytes
+
+    for (fs::directory_iterator it(walletdir); it != fs::directory_iterator(); ++it) {
+        if (!fs::is_regular_file(*it) || fs::is_symlink(*it)) continue;
+
+        std::ifstream file(it->path().string(), std::ios::binary);
```

Comment:
Note: to be compatible with #11687, you would drop the `continue` above and change this line to something like:

```c++
std::ifstream file(fs::is_regular_file(*it) ? it->path() : (it->path() / "wallet.dat")).string(), std::ios::binary);
```